src/common/services/session/session.service.js

```diff
@@ -46,95 +75,204 @@ const session = /* @ngInject */ function ($cookies, $rootScope, $http, $timeout,
+        })
+        .subscribe({
+          next: response => {
+            const data = response ? response.data : response
+            observer.next(data)
+            observer.complete()
+          },
+          error: err => {
+            observer.error(err)
+          }
+        })
+    })
   }
 
-  function signUp (email, password, first_name, last_name) {
-    // https://github.com/CruGlobal/cortex_gateway/wiki/Create-User
-    return Observable
-      .from($http({
-        method: 'POST',
-        url: casApiUrl('/register'),
-        withCredentials: true,
-        data: {
-          email: email,
-          password: password,
-          firstName: first_name,
-          lastName: last_name
+  function internalSignIn (lastPurchaseId) {
+    return new Observable(observer => {
+      authClient.isAuthenticated()
+        .then(isAuthenticated => {
+          if (!isAuthenticated) {
+            setRedirectingOnLogin()
+            return authClient.token.getWithRedirect()
+          }
+          return authClient.tokenManager.getTokens()
+        })
+        .then(tokens => {
+          if (!tokens) {
+            return
+          }
+          const data = { access_token: tokens.accessToken.accessToken }
+          // Only send lastPurchaseId if present and currently public
+          if (angular.isDefined(lastPurchaseId) && currentRole() === Roles.public) {
+            data.lastPurchaseId = lastPurchaseId
+          }
+          // Add marketing search queries back to URL once returned from Okta
+          const locationSearch = $window.sessionStorage.getItem(locationSearchOnLogin) || ''
+          if (locationSearch) {
+          // eslint-disable-next-line
+            const searchQueries = locationSearch.split(/\?|\&/);
+            $window.sessionStorage.removeItem(locationSearchOnLogin)
+            searchQueries.forEach((searchQuery) => {
+              const [search, value] = searchQuery.split('=')
+              if (search && value) {
+                $location.search(search, value)
+              }
+            })
+          }
+          removeOktaRedirectIndicator()
+          return $http({
+            method: 'POST',
+            url: oktaApiUrl('login'),
+            data: data,
+            withCredentials: true
+          })
+        })
+        .then(response => {
+          observer.next(response)
+          observer.complete()
+        })
+        .catch(err => {
+          observer.error(err)
+        })
+    })
+  }
+
+  function oktaIsUserAuthenticated () {
+    return Observable.from(authClient.isAuthenticated())
+  }
+
+  function signOut (redirectHome = true) {
+    return new Observable(observer => {
+      internalSignOut(redirectHome).subscribe({
+        next: response => {
+          observer.next(response)
+          observer.complete()
+        },
+        error: err => {
+          observer.error(err)
         }
-      }))
-      .map((response) => response.data)
+      })
+    })
   }
```

Comment:
Unless I'm missing something, the following is equivalent. If that is the case, we probably don't need separate `signOut` and `internalSignOut` functions.

```suggestion
  function signOut (redirectHome = true) {
    return internalSignOut(redirectHome)
  }
```